### ISSUE TITLE
Cannot delete property 'alwaysIncludePattern' ?

### ISSUE THREAD

**Issue by victor0801x** [nomiddlename/log4js-node#519]
(node: 8.2.1, log4js: 2.3.2)
I'm using stdout for development and dateFile for production.
but I got this error.
![2017-07-31 15-19-06](https://user-images.githubusercontent.com/4534038/28766990-9db9840c-7604-11e7-909a-2bba11d03698.png)
I seems options is readonly object.
I tried to fix by comment some codes, 
```js
// at streamroller/lib/DateRollingFileStream.js:37
/*
    delete options.alwaysIncludePattern;
    if (Object.keys(options).length === 0) {
      options = null;
    }
*/
```
for now works fine.
but I'm not sure is this actually hits the problem or will cause other potential errors ?

**Comment by nomiddlename** [comment]
Could you show me how you're configuring log4js, please? Is the config object you're passing to log4js some special read-only object?

**Comment by victor0801x** [comment]
this is my log4js config (in .json file)
```json
"log4js": {
    "appenders": {
        "out": {
            "type": "stdout",
            "pattern": "%[[%d{ISO8601}] %5p - (%X{_this}) %m%]"
        },
        "rollingFile": {
            "type": "dateFile",
            "filename": "./log/webLogs",
            "pattern": ".yyyy-MM-dd.log",
            "compress": true,
            "alwaysIncludePattern": true,
            "layout": {
                "type": "pattern",
                "pattern": "%[ [%d{ISO8601}] [%5p] (%X{_this}) - %m %]"
            }
        },
        "errors": {
            "type": "file",
            "filename": "./log/errors.log",
            "compress": true, "maxLogSize": 16777216, "backups": 3,
            "layout": {
                "type": "pattern",
                "pattern": "%[ [%d{ISO8601}] [%5p] (%X{_this}) - %m %]"
            }
        },
        "onlyError": { "type": "logLevelFilter", "appender": "errors", "level": "error" }
    },
    "categories": {
        "default": { "appenders": ["out", "rollingFile", "errors"], "level": "trace" },
        "trace": { "appenders": ["out", "rollingFile", "errors"], "level": "trace" },
        "debug": { "appenders": ["out", "rollingFile", "errors"], "level": "debug" },
        "info": { "appenders": ["out", "rollingFile", "errors"], "level": "info" }
    }
}
```
then, use config.js to read config
```js
const config = require('config');
const log4js = require('log4js');
log4js.configure(config.get('log4js'));
const webLogger = log4js.getLogger('trace');
webLogger.addContext('_this', path.parse(__filename).base);
```

**Comment by victor0801x** [comment]
I found config.get() use Object.defineProperty() and set writable : false, configurable: false

**Comment by nomiddlename** [comment]
Thanks for the info, that will help me with writing a test to cover it.

**Comment by nomiddlename** [comment]
Published to NPM in version 2.3.3
